payu/fsops.py

```diff
@@ -171,3 +173,41 @@ def is_conda():
     """Return True if python interpreter is in a conda environment"""
 
     return os.path.exists(os.path.join(sys.prefix, 'conda-meta'))
+
+
+def required_libs(bin_path):
+    """
+    Parses ldd output.
+    This function should only be called once per binary
+    i.e. Use a singleton pattern in the caller object.
+    PARAMETERS:
+        string bin_path: full path to the binary
+    RETURN:
+        dict: {filename-of-lib: fullpath-of-file}
+    """
+
+    # NOTE: ldd <binary>
+
+    # Example 1:
+    # libmpi.so.40 => /apps/openmpi/4.0.2/lib/libmpi.so.40 (0x00007fa493665000)
+    # libmpi_usempif08_Intel.so.40 => /apps/openmpi/4.0.2/lib/libmpi_usempif08_Intel.so.40 (0x00007fa492a7c000)
+    # libmpi_usempi_ignore_tkr_Intel.so.40 => /apps/openmpi/4.0.2/lib/libmpi_usempi_ignore_tkr_Intel.so.40 (0x00007fa492863000)
+    # libmpi_mpifh_Intel.so.40 => /apps/openmpi/4.0.2/lib/libmpi_mpifh_Intel.so.40 (0x00007fa4925cb000)
+
+    # Example 2:
+    # libmpi_usempif08.so.40 => $SPACKDIR/opt/spack/linux-rocky8-cascadelake/intel-2019.5.281/openmpi-4.1.5-ooyg5wc7sa3tvmcpazqqb44pzip3wbyo/lib/libmpi_usempif08.so.40 (0x00007f12671c0000)
+    # libmpi_usempi_ignore_tkr.so.40 => $SPACKDIR/opt/spack/linux-rocky8-cascadelake/intel-2019.5.281/openmpi-4.1.5-ooyg5wc7sa3tvmcpazqqb44pzip3wbyo/lib/libmpi_usempi_ignore_tkr.so.40 (0x00007f1266fb4000)
+    # libmpi_mpifh.so.40 => $SPACKDIR/opt/spack/linux-rocky8-cascadelake/intel-2019.5.281/openmpi-4.1.5-ooyg5wc7sa3tvmcpazqqb44pzip3wbyo/lib/libmpi_mpifh.so.40 (0x00007f1266d44000)
+    # libmpi.so.40 => $SPACKDIR/opt/spack/linux-rocky8-cascadelake/intel-2019.5.281/openmpi-4.1.5-ooyg5wc7sa3tvmcpazqqb44pzip3wbyo/lib/libmpi.so.40 (0x00007f12667c3000)
```

Comment:
In my opinion these examples should be moved into a test of `required_libs`.

Reply:
Ok, I've separating the parsing of ldd output into a separate file so its simpler to test. The other option could be to mock out the ldd call?